### DISCUSSION TITLE
ERROR FirebaseError: Missing or insufficient permissions.

### DISCUSSION THREAD

**Discussion by mulkrajsingh** [firebase/firebase-js-sdk#4599] [Q&A]
Now, before you suggest to change the rules.
It's an angular app and when the page is first loaded it shows the following error. But if I route to another tab and come back to the initial tab the data is fetched without any errors.

ERROR FirebaseError: Missing or insufficient permissions.
    at new n (http://localhost:4200/vendor.js:116763:23)
    at http://localhost:4200/vendor.js:125903:28
    at http://localhost:4200/vendor.js:125904:18
    at n.onMessage (http://localhost:4200/vendor.js:125937:10)
    at http://localhost:4200/vendor.js:125843:26
    at http://localhost:4200/vendor.js:125874:37
    at http://localhost:4200/vendor.js:121470:31
    at ZoneDelegate.invoke (http://localhost:4200/polyfills.js:377:30)
    at Zone.run (http://localhost:4200/polyfills.js:136:47)
    at http://localhost:4200/polyfills.js:870:40

I have browsed stackoverflow and youtube in search of an possible explanation and solution but everywhere the answer is same. That is to change the rules. But I am getting this error in both staging and production.

Thanks in advance for your time and effort.

[Edit]: Also when I route back to the initial tab, subscribe is called twice, thus making the total number of objects twice as original.

**Comment by b2m9** [comment]
Ok, I'm just guessing here. But without having more information I would say that when you query for the first time on load, your user might not be authenticated and thus your request is failing (maybe `currentUser` is `null`). By the time you query again, Firebase has a valid `currentUser` object and sends it along the next request, retrieving the data successfully.

**Reply by mulkrajsingh** [reply]
Thank-you. I will look into it. :)

**Comment by Feiyang1** [comment]
Are you using the alpha SDK, or the production SDK? What version number is it?

**Reply by mulkrajsingh** [reply]
it shows this in both of them

**Reply by akauppi** [reply]
@mulkrajsingh Are you using EcmaScript modules?

I had the same error message a while back, and I think it was caused by code in the body of a module, calling Firebase. It's recommended not to have side-effect code in the body, but... I do sometimes do that. :)

This tricks the programmer's brain because code no longer is executed linearly. There's an "import" sweep that happens before the code runs. And that sweep includes all the imported modules, and their submodules.

Hope this helps. For me, some restructuring took away the error.

---

Seeing that there's a `polyfills.js` in your call stack - are you the author of that or is it someone else?  What if you try without the polyfill..
